File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
  vector<int> twoSum(vector<int>& nums, int target) {
    map<int, int> cache;
    vector<int> result;
    for (vector<int>::size_type i = 0; i < nums.size(); i++) {
      if (cache.count(nums[i]) > 0) {
        result.push_back(cache[nums[i]]);
        result.push_back(i);
        return result;
      }
      cache[target - nums[i]] = i;
    }
    return result;
  }
};

File: 1-two-sum/1-two-sum.go
func twoSum(nums []int, target int) []int {
  cache := make(map[int]int)
  for i, v := range nums {
    if j, ok := cache[v]; ok {
      return []int{j, i}
    }
    cache[target - v] = i
  }
  return []int{}
}